src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -23,17 +23,17 @@ internal class ManagedIdentityClient
         internal const string MsiUnavailableError =
             "Authentication with managed identity is unavailable. No managed identity endpoint found.";
 
-        private readonly Lazy<ManagedIdentitySource> _identitySource;
+        private readonly ManagedIdentitySource _identitySource;
 
         public ManagedIdentityClient(RequestContext requestContext)
         {
-            _identitySource = new Lazy<ManagedIdentitySource>(() => SelectManagedIdentitySource(requestContext));
+            _identitySource = SelectManagedIdentitySource(requestContext);
```

Comment:
Can we please have a task to track the perf impact of caching / not caching the identity source?

Reply:
Created issue to track performance work: #3836 
Adding the code to log duration, will be helpful as we add more resources.